### PR TITLE
updated ERC-20 Approval Step

### PR DESCRIPTION
**Pull request by soft2dev**
I didn't make a contract change.
I updated the Approval Step for ERC20 tokens and added two buttons(Approve and Swap button) like uniswap and removed Approve step via Modal.

If you made a contract change, make sure to complete the checklist below before merging it in master.

Refer to our [documentation](https://github.com/OriginProtocol/security) for more details about contract security best practices.

Contract change checklist:
  - [ ] Code reviewed by 2 reviewers. See [template](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review.md) and [example](https://github.com/OriginProtocol/security/blob/master/templates/Contract-Code-Review-Example.md).
  - [ ] Unit tests pass
  - [ ] Slither tests pass with no warning
  - [ ] Echidna tests pass if PR includes changes to OUSD contract (not automated, run manually on local)

### CODE REVIEW

**Comment by sparrowDom** [comment]
Hey nice work! Have a couple of UX considerations: 
The allowance button should be shown only when the selected swap strategy does not have sufficient Allowance for the currently selected coin to swap. In the image below no swap strategy is yet selected
Screenshot 2021-12-20 at 10 25 46

When the button is shown the height of the UI on the right should be aligned
Screenshot 2021-12-20 at 10 26 26

On the button it says `Allow Origin Dollar to user your...` The `Origin Dollar` part of the text should change to: Curve, Uniswap V3, Uniswap V2.... depending on which swap strategy is currently selected.

File: dapp/src/components/buySell/DisclaimerTooltip.js [hunk 1]
```diff
@@ -34,7 +32,7 @@ const DisclaimerTooltip = ({
         {children && (
           <div
             className="d-flex align-items-center justify-content-center"
-            onClick={(e) => {
+            onMouseOver={(e) => {
```
Comment (sparrowDom) [inline dapp/src/components/buySell/DisclaimerTooltip.js:35]: This will probably not work on mobile. Would be cool that it works in both envrionments

Comment (sparrowDom) [reply dapp/src/components/buySell/DisclaimerTooltip.js:35]: Also looks like we haven't addressed the z-sorting well in the past: 
Screenshot 2021-12-20 at 10 41 25

File: dapp/src/components/buySell/SwapHomepage.js [hunk 2]
```diff
@@ -611,9 +640,69 @@ const SwapHomepage = ({
           </a>
           <button
             //disabled={formHasErrors || buyFormHasWarnings || !totalOUSD}
+            className={`btn-blue pl-2 pr-2 buy-button mt-2 mt-md-0 w-100 ${
```
Comment (sparrowDom) [inline dapp/src/components/buySell/SwapHomepage.js:643]: There is quite a lot of logic in the approve button component, could we separate that into another component? The SwapHomepage is already too big -> which is something I am guilty of and should make it smaller in the future.

**Review by sparrowDom — COMMENTED** [review]
Great job,  could you also address the couple of minor comments inline please?

File: dapp/src/components/buySell/SwapHomepage.js (hunk 2 above)
Comment (soft2dev) [reply dapp/src/components/buySell/SwapHomepage.js]: Made SwapButton component.

File: dapp/src/components/buySell/DisclaimerTooltip.js (hunk 1 above)
Comment (soft2dev) [reply dapp/src/components/buySell/DisclaimerTooltip.js]: Fixed UI issues, used onClick event for the tooltip, shows only one button as the status(swap or approve)

**Pull request closed without merging**